static/app/components/segmentedControl.tsx

```diff
@@ -153,7 +153,8 @@ function Segment({
 
 const GroupWrap = styled('div')<{priority: Priority; size: FormSize}>`
   position: relative;
-  display: inline-flex;
+  display: inline-grid;
+  grid-auto-flow: column;
```

Comment:
Use grid display to automatically stretch segments out to fill the outer container, if the container is wider than its contents.

**Before ——**
<img width="556" alt="Screenshot 2023-02-01 at 10 03 59 AM" src="https://user-images.githubusercontent.com/44172267/216125880-c9d34c77-71b0-4025-b824-8c7d37b4ab85.png">

**After ——**
<img width="556" alt="Screenshot 2023-02-01 at 10 03 40 AM" src="https://user-images.githubusercontent.com/44172267/216125808-7e4f7262-a2f9-4e4d-8476-f47fc67d8093.png">

This only applies when `SegmentedControl`'s width is being controlled by either restyling or being inside a flex/grid container. By default the container will "hug" its contents:
<img width="225" alt="Screenshot 2023-02-01 at 10 06 14 AM" src="https://user-images.githubusercontent.com/44172267/216126306-6dbd1c5d-a9db-4b9f-aa13-4d4aa9136cf7.png">